lib/parsers/wav_parser.rb

```diff
@@ -0,0 +1,102 @@
+class FormatParser::WAVParser
+  include FormatParser::IOUtils
+
+  def information_from_io(io)
+    io.seek(0)
+
+    # Read the RIFF header. Chunk descriptor should be RIFF, the size should 
+    # contain the size of the entire file in bytes minus 8 bytes for the 
+    # two fields not included in this count: chunk_id and size.
+    chunk_id, size, riff_type = safe_read(io, 12).unpack('a4la4')
+
+    # The chunk_id and riff_type should be RIFF and WAVE respectively
+    return unless chunk_id == 'RIFF' && riff_type == 'WAVE'
+
+    # There are no restrictions upon the order of the chunks within a WAVE file,
+    # with the exception that the Format chunk must precede the Data chunk.
+    # The specification does not require the Format chunk to be the first chunk
+    # after the RIFF header.
+    # http://soundfile.sapp.org/doc/WaveFormat/
+    # For WAVE files containing PCM audio format we parse the 'fmt ' and
+    # 'data' chunks while for non PCM audio formats the 'fmt ' and 'fact'
+    # chunks. In the latter case the order fo appearence of the chunks is
+    # arbitrary.
+    fmt_processed = false
+    fact_processed = false
+    fmt_data = {}
+    total_sample_frames = 0
+    loop do
+      chunk_type, chunk_size = safe_read(io, 8).unpack('a4l')
+      case chunk_type
+      when 'fmt '
```

Comment:
Can we have a comment or an escape character here so it's clear that the space is deliberate?

Reply:
I agree, it's a bit strange to have a space character in the ID of one of the most basic chunk types. I added a comment to draw the eye.